src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -73,50 +82,100 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
         }
 
         public void AddNew() {
-            SelectConnection(new ConnectionViewModel());
+            _shell.AssertIsOnMainThread();
+            CancelEdit(SelectedConnection);
+            NewConnection = new ConnectionViewModel();
+        }
+        
+        public void CancelEdit(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            if (connection == NewConnection) {
+                NewConnection = null;
+            } else if (connection != null) {
+                connection.Reset();
+                connection.IsEditing = false;
+            }
+        }
+
+        public void BrowseLocalPath(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            string latestLocalPath;
+            Uri latestLocalPathUri;
+
+            if (connection.Path != null && Uri.TryCreate(connection.Path, UriKind.Absolute, out latestLocalPathUri) && latestLocalPathUri.IsFile && !latestLocalPathUri.IsUnc) {
+                latestLocalPath = latestLocalPathUri.LocalPath;
+            } else { 
+                latestLocalPath = Environment.SystemDirectory;
+
+                try {
+                    latestLocalPath = new RInstallation().GetCompatibleEnginePathFromRegistry();
```

Comment:
if it came from ChooseRFolderUIEditor class, then that class can be deleted I think